puppet/zulip/files/postgresql/process_fts_updates

```diff
@@ -80,51 +83,50 @@ def am_master(cursor: psycopg2.extensions.cursor) -> bool:
     return not cursor.fetchall()[0][0]
 
 
-pg_args = {}
+def get_config(
+    config_file: configparser.RawConfigParser,
+    section: str,
+    key: str,
+    default_value: str = "",
+) -> str:
+    if config_file.has_option(section, key):
+        return config_file.get(section, key)
+    return default_value
+
+
+pg_args = {
+    "connect_timeout": "600",
+    "user": "zulip",
+    "dbname": "zulip",
+}
 
-# Path to the root of the Zulip codebase in production
-sys.path.insert(0, "/home/zulip/deployments/current")
-# Path to the root of the Zulip codebase in development
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../..")))
+USING_PGROONGA = False
 try:
+    # Case (1); we insert the path to the development root.
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../..")))
+    # Case (2) could take this branch if it set up the venv first, but
+    # there is no reason to unnecessarily bring in the Django
+    # dependency; it will fall into the path below due the venv not
+    # being configured.
     os.environ["DJANGO_SETTINGS_MODULE"] = "zproject.settings"
     from django.conf import settings
 
-    if settings.REMOTE_POSTGRES_HOST != "":
-        pg_args["host"] = settings.REMOTE_POSTGRES_HOST
-    if settings.REMOTE_POSTGRES_PORT != "":
-        pg_args["port"] = settings.REMOTE_POSTGRES_PORT
+    pg_args["host"] = settings.DATABASES["default"]["HOST"]
+    pg_args["port"] = settings.DATABASES["default"].get("PORT")
+    pg_args["password"] = settings.DATABASES["default"].get("PASSWORD")
+    pg_args["sslmode"] = settings.DATABASES["default"]["OPTIONS"].get("sslmode")
```

Comment:
I think it might be important to avoid putting empty string values in `pg_args`, as the previous code did with its conditionals.  Unless we've explicitly verified that e.g. `port=""` does something reasonable.  

It's annoying to have to think about that change in any case when reading this complex reordering commit, so better would be to just preserve the old logic.

Reply:
@alexmv would you be okay tackling this bit?

Reply:
Split into a bunch of commits.